pkg/network/node/egressip.go

```diff
@@ -284,37 +295,46 @@ func (eip *egressIPWatcher) deleteNamespaceEgress(vnid uint32) {
 	eip.updateNamespaceEgress(vnid, nil)
 }
 
-func (eip *egressIPWatcher) syncEgressIPs() {
-	changedEgressIPs := make(map[*egressIPInfo]bool)
-	for _, eg := range eip.changedEgressIPs {
-		changedEgressIPs[eg] = true
+func (eip *egressIPWatcher) egressIPActive(eg *egressIPInfo) (bool, error) {
+	if len(eg.nodes) == 0 || len(eg.namespaces) == 0 {
+		return false, nil
 	}
-	eip.changedEgressIPs = eip.changedEgressIPs[:0]
-
-	changedNamespaces := make(map[*namespaceEgress]bool)
-	for _, ns := range eip.changedNamespaces {
-		changedNamespaces[ns] = true
+	if len(eg.nodes) > 1 {
+		return false, fmt.Errorf("Multiple nodes (%s, %s) claiming EgressIP %s", eg.nodes[0].nodeIP, eg.nodes[1].nodeIP, eg.ip)
 	}
-	eip.changedNamespaces = eip.changedNamespaces[:0]
+	if len(eg.namespaces) > 1 {
+		return false, fmt.Errorf("Multiple namespaces (%d, %d) claiming EgressIP %s", eg.namespaces[0].vnid, eg.namespaces[1].vnid, eg.ip)
+	}
+	for _, ip := range eg.namespaces[0].requestedIPs {
+		eg2 := eip.egressIPs[ip]
+		if eg2 != eg && len(eg2.nodes) == 1 && eg2.nodes[0] == eg.nodes[0] {
+			return false, fmt.Errorf("Multiple EgressIPs (%s, %s) for VNID %d on node %s", eg.ip, eg2.ip, eg.namespaces[0].vnid, eg.nodes[0].nodeIP)
```

Comment:
@danwinship What is the reason to make the egressIP unavailable when all the egressIPs which are claimed by namespace are held by the same node? IMO, this condition will break the HA, but should be harmless if only let the first element work.

Reply:
Ok, that makes sense. Thanks.